bin/generate.py

```diff
@@ -532,47 +541,92 @@ def __init__(self, problem, generator_config, key, name: str, yaml, parent):
         else:
             self.hash = combine_hashes(self.hash)
 
-        # Error for listed cases with identical input.
+        self.copy_of = None
         if self.hash in generator_config.rules_cache:
-            # This is fatal to prevent crashes later on when running generators in parallel.
-            # https://github.com/RagnarGrootKoerkamp/BAPCtools/issues/310
-            self.parse_error = (
-                f'Found identical input at {generator_config.rules_cache[self.hash]}. Skipping.'
-            )
-            generator_config.n_parse_error += 1
-            return
-        generator_config.rules_cache[self.hash] = self.path
+            self.copy_of = generator_config.rules_cache[self.hash]
+        else:
+            generator_config.rules_cache[self.hash] = self
 
     def generate(t, problem, generator_config, parent_bar):
         bar = parent_bar.start(str(t.path))
 
         t.generate_success = False
 
+        if t.copy_of is not None and t.count_index == 0:
+            bar.warn(
+                f'Found identical rule at {t.copy_of.path}. Use "count: <int>" if you want multiple identical testcases.'
+            )
+
         # Some early checks.
+        if t.copy_of is not None and not t.copy_of.generate_success:
+            bar.done(False, f'See {t.copy_of.path}')
+            return
         if t.parse_error is not None:
             bar.done(False, t.parse_error)
             return
         if t.generator and t.generator.program is None:
             bar.done(False, f'Generator didn\'t build.')
             return
+        if t.hash is None:
+            # Input can only be missing when the `copy:` does not have a corresponding `.in` file.
+            # (When `generate:` or `in:` is used, the input is always present.)
+            bar.done(False, f'{t.copy} does not exist.')
+            return
 
         target_dir = problem.path / 'data' / t.path.parent
         target_infile = target_dir / (t.name + '.in')
         target_ansfile = target_dir / (t.name + '.ans')
 
-        # Input can only be missing when the `copy:` does not have a corresponding `.in` file.
-        # (When `generate:` or `in:` is used, the input is always present.)
-        if t.hash is None:
-            bar.done(False, f'{t.copy} does not exist.')
-            return
-
         # E.g. bapctmp/problem/data/<hash>.in
         cwd = problem.tmpdir / 'data' / t.hash
         cwd.mkdir(parents=True, exist_ok=True)
         infile = cwd / 'testcase.in'
         ansfile = cwd / 'testcase.ans'
         meta_path = cwd / 'meta_.yaml'
 
+        # Returns False when some files were skipped.
+        def copy_generated():
```

Comment:
Do you want to move this into `def generate_includes()` below? Since you comment on the parallel invocation that includes and copies happen in step 2, after dirs and testcases?

Reply:
i dont know what you mean? ^^'